src/light.cpp

```diff
@@ -46,16 +46,20 @@ void set_light_table(float gamma)
 // Gamma correction
 	gamma = rangelim(gamma, 0.5f, 3.0f);
 
-	for (size_t i = 0; i < LIGHT_SUN; i++) {
+// Boundary values should be fixed
+	light_LUT[0] = 0;
```

Comment:
I'm not sure about this, the midboost can cause light level 0 to have non zero visual brightness, and this might be useful to have in some debugging or dev situations.
The smoothness of the lightcurve at the dark end is essential and critical for the world to look right, forcing zero will break this smoothness.

Reply:
See the history of light curves, many have non-zero brightness for light level 0:
https://github.com/minetest/minetest/blob/231ac33d34dfaaddf292c5f31b1eae43eeefba2d/src/light.cpp#L186
So i disapprove of `light_LUT[0] = 0;`

Reply:
Also see https://github.com/minetest/minetest/pull/6639#issuecomment-346515132
Our current light curve is non-zero (actually 2) at light level 0.

Reply:
> this will be useful to set in debugging or dev situations.

...and to gain fullbright easily.

> The smoothness of the lightcurve

There is no smoothness, the table is discrete.

> have non-zero brightness for light level 0

That’s a bug

> `// Use for debugging in dark `

That’s like keeping `printf("I'm here %d", iter)` in a release.

Reply:
> and to gain fullbright easily.

Ahh good point.
By smoothness i mean the consistency of the light curve even though it is discrete.
No it's not a bug, it's a preference, see the history of light curves in MT, many are non-zero brightness at light level zero. However the fullbright issue seems to override this.

So i'm fine with this as it is.